src/test/java/com/google/devtools/build/lib/rules/cpp/CcLibraryConfiguredTargetTest.java

```diff
@@ -1633,4 +1633,84 @@ public void testPrecompiledFilesFromDifferentConfigs() throws Exception {
         ")");
     checkError("//foo", "Trying to link twice");
   }
+
+  @Test
+  public void testDynamicLinkTwiceAfterTransition() throws Exception {
```

Comment:
This test is great. Can you also add one that makes sure it works to link a library with a transition applied to it (i.e. different config) if it's unique? Also for completeness add another dependency to the cc_binary, just to know for sure that we have a mix for libraries in one config and libraries in a different config being linked into the same binary.